lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
+                reject(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: error,
+                    });
+                return;
+            });
+        });
+    });
+}
+
+/**
+ * Collect analytics data from the runtime system
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Object} Analytics payload
+ */
+function fetchPlatformData(context, eventName) {
     const os = require('os');
-    const request = require('request');
     const { machineId } = require("node-machine-id");
+    const environment = utils.getContext();
+
+    return new Promise((resolve, reject) => {
+        machineId()
+        .catch((e) => {
+            reject('Analytics submission failed');
+            return;
+        })
+
+        .then((identifier) => {
+            if (!identifier) {
+                identifier = sha256('unknown');
+            }
 
-    module.exports = function(eventName, context) {
-        // Submit analytics after some time - we do this to give the application a chance to disable analytics.
+            // payloads for webhook and MixPanel differ slightly
+            const payload = {
+                'webHook': {
+                    'event': eventName,
+                    'properties': {
+                        'token': 'aab85907a13e1ff44a95be539d9942a9',
+                        'distinct_id': identifier,
+                        'Anonymized Machine Identifier': identifier,
+                        'Anonymized Application ID': sha256(__dirname),
+                        'Binding': 'javascript',
+                        'Target': environment,
+                        'Version': context.version,
+                        'Language': 'javascript',
+                        'OS Type': os.platform(),
+                        'OS Version': os.release(),
+                        'Node.js versions': process.versions
+                    }
+                },
+                'mixPanel': {
+                    'event': eventName,
+                    'properties': {
+                        'token': 'aab85907a13e1ff44a95be539d9942a9',
+                        'distinct_id': identifier,
+                        'Anonymized Machine Identifier': identifier,
+                        'Anonymized Application ID': sha256(__dirname),
+                        'Binding': environment,
+                        'Version': context.version,
+                        'Language': 'javascript',
+                        'OS Type': os.platform(),
+                        'OS Version': os.release(),
+                        'Node.js versions': process.versions
+                    }
+                }
+            };
+
+            resolve(payload);
+            return;
+        });
```

Comment:
```suggestion
        }, reject);
```

If you don't move the `.catch` call down here, you need to pass `reject` into this call to `.then` or you risk an unhandled promise rejection.

Reply:
`.catch` has been duly moved.